### DISCUSSION TITLE
How to connect, authenticate and save PB state in Svelte store

### DISCUSSION THREAD

**Discussion by g1sbi** [pocketbase/pocketbase#2178] [Q&A] [answered]
What I'm trying to do is connect to PB, authenticate and then save the authenticated state in a store, such that I'd be able to use it in any file without having to reauthenticate everytime. What I tried was this:

```
import { writable } from 'svelte/store';
import PocketBase from 'pocketbase';

export const pb = writable();

async function authenticate() {
	const pbInstance = new PocketBase('http://localhost:8090');
	await pbInstance.admins.authWithPassword('example@gmail.com', 'unsafepassword');
	pb.set(pbInstance);
}

authenticate();
```

But when I import `pb` in another file and try to use a method, like `pb.collection` I get an error saying "Property 'collection' does not exist on type 'Writable<unknown>'"

**Comment by ganigeorgiev** [comment, marked as answer]
I don't understand why do you need a store for this.

If you want to maintain a single global instance, you can just place the JS SDK initialization in its own file and import the exported instance, something like:

```js
const pb = new PocketBase('http://localhost:8090');

await pb.admins.authWithPassword('example@gmail.com', 'unsafepassword');

export default pb;
```

If for some reason you still want to keep the instance in a store, after importing it you have to access it with a `$` prefix, (aka. `$pb.*`). For more info check https://svelte.dev/docs#component-format-script-4-prefix-stores-with-$-to-access-their-values.

**Reply by g1sbi** [reply]
I didn't realize I was overcomplicating this. This works fine, thank you!

**Comment by tastycrayon** [comment]
You can see an example implementation here that uses both email and google oauth
https://github.com/tastycrayon/sveltekit-pocketbase-auth
